parastell/radial_distance_utils.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+from . import magnet_coils
+from . import invessel_build as ivb
+import pystell.read_vmec as read_vmec
+
+import cubit
+
+
+def calc_z_radius(point):
+    """
+    Calculate the distance from the z axis.
+
+    Arguments:
+        point (iterable of [x,y,z] float): point to find distance for
+    Returns:
+        (float): distance to z axis.
+    """
+    return (point[0] ** 2 + point[1] ** 2) ** 0.5
+
+
+def get_start_index(filament):
+    """
+    Find the index at which the filament crosses the xy plane on the OB
+    side of the filament
+
+    Arguments:
+        filament (list of list of float): List of points defining the filament
+
+    Returns:
+        max_z_index (int): index at which the filament crosses the xy plane
+    """
+    max_z_index = None
+    max_z_radius = 0
+    for index, (point, next_point) in enumerate(
+        zip(filament[0:-1], filament[1:])
+    ):
+        if point[2] / next_point[2] < 0:
+            z_radius = calc_z_radius(point)
+            if max_z_radius < z_radius:
```

Comment:
What if the max z point isn't where the filament crosses the x-y plane?

Reply:
Seems like this is a poor variable name. It's called max_z_radius because its should end up being the point where it crosses the x-y plane with the larger distance to the z axis (has the larger radius if the center is the z axis) of the two crossings, not anything to do with the actual z value of the points. The if statement above checks for a sign change in z between the two points, and only checks for what the z radius is if there is a sign change (crosses the xy plane). I'll rename these to something hopefully a bit clearer.